### PR TITLE
Support SMISMEMBER

### PR DESCRIPTION
**Pull request by Avital-Fine**
https://redis-stack.io/commands/smismember/

#2055

### CODE REVIEW

File: src/StackExchange.Redis/Interfaces/IDatabase.cs [hunk 1]
```diff
@@ -1102,6 +1102,19 @@ public interface IDatabase : IRedis, IDatabaseAsync
...
+        /// <summary>
+        /// Returns whether each member is a member of the set stored at key.
+        /// </summary>
+        /// <param name="key">The key of the set.</param>
+        /// <param name="values">The members to check for.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>
+        /// <see langword="true"/> if the element is a member of the set.
+        /// <see langword="false"/> if the element is not a member of the set, or if key does not exist.
+        /// </returns>
```
Comment (NickCraver) [inline src/StackExchange.Redis/Interfaces/IDatabase.cs:1114]: In this case doc should say an _array_ of these true/false, matching position, e.g. 
```suggestion
        /// <returns>
        /// An array of booleans corresponding to <paramref name="values"/>, for each:
        /// <see langword="true"/> if the element is a member of the set.
        /// <see langword="false"/> if the element is not a member of the set, or if key does not exist.
        /// </returns>
```

What happens when the key isn't present, all falses?

File: tests/StackExchange.Redis.Tests/Sets.cs [hunk 2]
```diff
@@ -38,6 +38,31 @@ public void SScan()
             }
         }
 
+        [Fact]
+        public void SetContains()
+        {
+            using (var conn = Create())
```
Comment (NickCraver) [inline tests/StackExchange.Redis.Tests/Sets.cs:44]: New `using` style on tests please! I'll add this to `.editorconfig` as a stronger suggestion so it's automatic.

File: tests/StackExchange.Redis.Tests/Sets.cs [hunk 3]
```diff
@@ -38,6 +38,31 @@ public void SScan()
...
+                }
+
+                // Single member
+                var isMemeber = db.SetContains(key, 1);
+                Assert.True(isMemeber);
+
+                // Multi members
+                var areMemebers = db.SetContains(key, new RedisValue[] {1, 2, 3});
+                Assert.Equal(3, areMemebers.Length);
+                Assert.True(areMemebers[1]);
```
Comment (NickCraver) [inline tests/StackExchange.Redis.Tests/Sets.cs:62]: We should test that `false` works here too :)

File: src/StackExchange.Redis/ResultProcessor.cs [hunk 4]
```diff
@@ -1262,6 +1265,22 @@ protected override bool SetResultCore(PhysicalConnection connection, Message mes
             }
         }
 
+        private sealed class BooleanArrayProcessor : ResultProcessor<Boolean[]>
```
Comment (NickCraver) [inline src/StackExchange.Redis/ResultProcessor.cs:1268]: Use C# types please:
```suggestion
        private sealed class BooleanArrayProcessor : ResultProcessor<bool[]>
```

File: src/StackExchange.Redis/ResultProcessor.cs [hunk 5]
```diff
@@ -1262,6 +1265,22 @@ protected override bool SetResultCore(PhysicalConnection connection, Message mes
...
+        {
+            protected override bool SetResultCore(PhysicalConnection connection, Message message, in RawResult result)
+            {
+                switch (result.Type)
+                {
+                    case ResultType.MultiBulk:
+                        var arr = result.GetItemsAsBooleans()!;
```
Comment (NickCraver) [inline src/StackExchange.Redis/ResultProcessor.cs:1275]: This assertion is blind, so a null result would go boom - be sure to add a check, e.g.:
```suggestion
                    case ResultType.MultiBulk when !result.IsNull:
                        var arr = result.GetItemsAsBooleans()!;
```

But I'm not sure this is correct. We should consider the case the key is missing, and add tests for that in a few of these PRs to ensure we get the expected result.

File: tests/StackExchange.Redis.Tests/Sets.cs (hunk 2 above)
Comment (NickCraver) [reply tests/StackExchange.Redis.Tests/Sets.cs:44]: These also need some v6_2_0 checking love!

**Review by NickCraver — CHANGES_REQUESTED** [review]
Spotted a few things cross-PR in reviewing these, notes inline! Thanks again, these PRs are fantastic and you're cranking 'em out <3

File: tests/StackExchange.Redis.Tests/Sets.cs (hunk 2 above)
Comment (Avital-Fine) [reply tests/StackExchange.Redis.Tests/Sets.cs]: Is that the new style that you meant?

Comment (Avital-Fine) [reply tests/StackExchange.Redis.Tests/Sets.cs]: If it's right I can fix the style in the other tests in this file :)

Comment (Avital-Fine) [reply tests/StackExchange.Redis.Tests/Sets.cs]: fixed :)

File: src/StackExchange.Redis/Interfaces/IDatabase.cs (hunk 1 above)
Comment (Avital-Fine) [reply src/StackExchange.Redis/Interfaces/IDatabase.cs]: yes

File: tests/StackExchange.Redis.Tests/Sets.cs [hunk 6]
```diff
@@ -14,234 +14,244 @@ public class Sets : TestBase
...
+            var key = Me();
 
-                db.KeyDelete(key, CommandFlags.FireAndForget);
+            db.KeyDelete(key, CommandFlags.FireAndForget);
 
-                var arr = db.SetPop(key, 1);
-                Assert.Empty(arr);
-            }
+            var arr = db.SetPop(key, 1);
+            Assert.Empty(arr);
```
Comment (NickCraver) [inline tests/StackExchange.Redis.Tests/Sets.cs:254]: Let's please only use the new syntax in the PRs (doing as little diff as possible) - I'll do a separate cleanup pass that's much cleaner overall git history. If we do these mass cleanups mixed in other PRs, history is very confusing later on.

**Review by NickCraver — APPROVED** [review]
Looking good - thank you for this!

**Comment by NickCraver** [comment]
Pushed one last tidy on the tests, we should nuke the key when starting to ensure a consistent test run - after that clears will merge in.

**Pull request merged**